pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -100,6 +102,17 @@ func NewCollector(cfg *Config, statesInformer statesinformer.StatesInformer, met
 	if c.config == nil {
 		c.config = NewDefaultConfig()
 	}
+	if ready := initNvml(); ready {
+		// init gpu device.
+		c.context.gpuDeviceManager = &gpuDeviceManager{}
+		if err := c.context.gpuDeviceManager.InitGPUData(); err != nil {
+			klog.Errorf("failed to init gpu device, err: %v", err)
```

Comment:
can we separate the scenario of `no gpu`(info) and `gpu init failed`(error)
and maybe for some `nvml.error`, initialization can be tried during later collections without reboot koordlet. consider adding a TODO comments here if the implementation is complex